tools/codegen/dest/lazy_ir.py

```diff
@@ -180,17 +182,21 @@ def lazy_tensor_decls(value_types: List[NamedCType], tensor_class: str, schema:
         if t.name in schema.wrapped_scalar_names:
             # no lazy tensor wrapper for scalars that are promoted to IR values
             continue
-        if isinstance(t.type, BaseCType):
-            lazy_tensor_decls.append(
-                f"{tensor_class}Ptr lazy_{t.name} = "
-                f"torch::lazy::GetLtcTensorOrCreateForWrappedNumber({t.name}, *common_device);")
+        elif isinstance(t.type, BaseCType):
+            if t.type.type is tensorListValueT:
```

Comment:
Going back to Nick's question [above](https://github.com/pytorch/pytorch/pull/73793/files#r822284536), I think Will's comment [here](https://github.com/pytorch/pytorch/pull/73793/files#diff-97a6e7faf841231c6814dbe6eebbb10217c557cf1437ffa241dca5d6cb10cfeeR11) makes sense to me. It looks like you're using the `NamedCType` here for two purposes:

(1) getting the C++ string to emit (`torch::lazy::Value`)
(2) pattern matching on the type to figure out extra stuff to emit (it's a TensorList type, so emit extra stuff)

But `NamedCType` isn't great for pattern matching, and `tensorListValueT` looks like a good example of why - when we see `Tensor` and `Tensor[]` arguments in the schema, we want both to map to a `torch::lazy::Value` (so they're C++ type is the same), but we still need to handle those cases differently (which you get around by creating a separate `CType` that you can pattern match on for Tensor lists, even though the C++ type is the same).

This totally doesn't require an immediate change (I saw the PR already landed, just figured I'd add a quick comment since I was tagged :) ). One.. slightly less hacky / easy way to get rid of the `tensorListValueT` hack would be to make `LazyIrSchema` hold a `Tuple[Binding]` instead of a `Tuple[NamedCType]`. That's still a bit weird since that's not the main purpose of `Binding`, but if you look at where it's [defined](https://github.com/pytorch/pytorch/blob/aff74cd654571ae49c650d124861821b90660251/tools/codegen/api/types.py#L305), it holds both a `NamedCType` (where you can store `valueT`) *and* an `Argument` (that can hold the `Tensor[]` information, which you can use for pattern matching).

The change for `LazyIrSchema`'s constructor would be pretty simple - something like:
```
...
positional_arg_types.append(Binding(
            nctype=NamedCType(arg.name, process_ir_type(arg.type)),
            name=arg.name,
            default=None,
            argument=arg,
))
...
```